packages/accordions/src/elements/accordion/Accordion.tsx

```diff
@@ -5,46 +5,66 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { useRef, useEffect, forwardRef, useMemo } from 'react';
+import React, { Children, forwardRef, useMemo } from 'react';
 import { useAccordion } from '@zendeskgarden/container-accordion';
 import { IAccordionProps } from '../../types';
 import { StyledAccordion } from '../../styled';
-import { AccordionContext } from '../../utils';
-import { Section } from '../accordion/components/Section';
-import { Header } from '../accordion/components/Header';
-import { Label } from '../accordion/components/Label';
-import { Panel } from '../accordion/components/Panel';
+import { AccordionContext, SectionContext } from '../../utils';
+import { Section } from './components/Section';
+import { Header } from './components/Header';
+import { Label } from './components/Label';
+import { Panel } from './components/Panel';
 
 const AccordionComponent = forwardRef<HTMLDivElement, IAccordionProps>(
   (
     {
-      level,
+      children,
       isBare,
-      onChange,
       isCompact,
       isAnimated,
       isExpandable,
       isCollapsible,
+      level,
+      onChange,
       defaultExpandedSections,
       expandedSections: controlledExpandedSections,
       ...props
     },
     ref
   ) => {
+    const { sections, sectionChildren } = useMemo(
+      () =>
+        Children.toArray(children)
+          .map((child, index) => (
+            <SectionContext.Provider key={index} value={index}>
+              {child}
+            </SectionContext.Provider>
```

Comment:
I'd like to hear a little bit more about this children wrapping vs. the previous context setup for this component. Should there be concerns about a more brittle DOM structure that is unable to handle intervening nodes?

Also, if this sticks, can `isValidElement` be used to eliminate the `any` casting? Ref: https://github.com/zendeskgarden/react-components/blob/3b332883cfe7fd23d285f7358a4b671c83fa3948/packages/dropdowns.next/src/elements/combobox/utils.ts#L38-L67

Reply:
The idea here is that we expect:
```jsx
<accordion>
  <section />
  <section />
  <div>
    <section />
  </div>
</accordion>
```
would give each group the context it needs. This of course relies on the children to take this into account. It actually provides more flexibility when it comes to unique DOM structure requirements, so long as the top level children follow the given pattern.